vm/systemSmartContracts/delegationManager.go

```diff
@@ -97,29 +113,262 @@ func (d *delegationManager) Execute(args *vmcommon.ContractCallInput) vmcommon.R
 		return d.init(args)
 	case "createNewDelegationContract":
 		return d.createNewDelegationContract(args)
-	case "deleteDelegationContract":
-		return d.deleteDelegationContract(args)
 	case "getAllContractAddresses":
 		return d.getAllContractAddresses(args)
+	case "changeBaseIssuingCost":
+		return d.changeBaseIssuingCost(args)
+	case "changeMinDeposit":
+		return d.changeMinDeposit(args)
 	}
 
+	d.eei.AddReturnMessage("invalid function to call")
 	return vmcommon.UserError
 }
 
-func (d *delegationManager) getAllContractAddresses(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+func (d *delegationManager) init(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if args.CallValue.Cmp(zero) != 0 {
+		d.eei.AddReturnMessage("callValue must be 0")
+		return vmcommon.UserError
+	}
+
+	managementData := &DelegationManagement{
+		NumberOfContract: 0,
+		LastAddress:      vm.FirstDelegationSCAddress,
+		MinServiceFee:    d.minFee,
+		MaxServiceFee:    d.maxFee,
+		BaseIssueingCost: d.baseIssuingCost,
+		MinDeposit:       d.minCreationDeposit,
+	}
+	err := d.saveDelegationManagementData(managementData)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	delegationList := &DelegationContractList{Addresses: make([][]byte, 0)}
+	err = d.saveDelegationContractList(delegationList)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
 	return vmcommon.Ok
 }
 
-func (d *delegationManager) deleteDelegationContract(_ *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+func (d *delegationManager) createNewDelegationContract(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	err := d.eei.UseGas(d.gasCost.MetaChainSystemSCsCost.DelegationMgrOps)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.OutOfGas
+	}
+	if d.callerAlreadyDeployed(args.CallerAddr) {
+		d.eei.AddReturnMessage("caller already deployed a delegation sc")
+		return vmcommon.UserError
+	}
+
+	delegationManagement, err := d.getDelegationManagementData()
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	minValue := big.NewInt(0).Add(delegationManagement.MinDeposit, delegationManagement.BaseIssueingCost)
+	if args.CallValue.Cmp(minValue) < 0 {
+		d.eei.AddReturnMessage("not enough call value")
+		return vmcommon.UserError
+	}
+
+	delegationList, err := d.getDelegationContractList()
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	depositValue := big.NewInt(0).Sub(args.CallValue, delegationManagement.BaseIssueingCost)
```

Comment:
what happens with the BaseIssueingCost value? Where does it go?

Reply:
it is still in discussion. It stays in this contract for now.

Reply:
Please add a TODO to revise this and not forget about it.